public/javascripts/EditCartItems.js

```diff
@@ -0,0 +1,290 @@
+/**
+ * Created by johnschroeder on 6/23/15.
+ */
+
+//TODO: make sure the database hit works once stored procedures stop timing out.
+//TODO: comment!
+var editCartItems = {
+    model: {products: [], productIDList: []},
+    cartID: parseInt(window.args.cartID),
+    cartName: window.args.cartName,
+    carts: [],
+    productTable: [],
+    init: function() {
+        $.get(window.apiRoute + "/Carts/GetCartItems/" + this.cartID, function(res) {
+            if (res && res.length) {
+                editCartItems.cartItems = JSON.parse(res)[0];
+                editCartItems.buildProductIDList();
+            } else {
+                $("#response").text("Error: EditCartItems.init: No response.");
+            }
+        }).fail(function(res) {
+            $("#response").text("Error: EditCartItems.init: Connection error.");
+        });
+    },
+
+    buildProductIDList: function() {
+        $("#cart_name").text(this.cartName);
+
+        for (var i = 0; i < this.cartItems.length; ++i) {
+            var index = this.cartItems[i].ProductID;
+            var model = editCartItems.model;
+            var productTable = editCartItems.model.products;
+            if (typeof editCartItems.productTable[index] === 'undefined') {
+                productTable[index] = {};
+                productTable[index].productID = index;
+                model.productIDList.push(index);
+                productTable[index].productName = this.cartItems[i].ProductName;
+                productTable[index].editing = false;
+                productTable[index].rows = [];
+                productTable[index].sizes = [];
+
+                editCartItems.productTable[index] = [];
+                editCartItems.productTable[index].push({
+                    name: this.cartItems[i].ProductName,
+                    totalQuantity: 0,
+                    productID: index
+                });
+            }
+            editCartItems.productTable[index].push(this.cartItems[i]);
+            editCartItems.productTable[index][0].totalQuantity += this.cartItems[i].Total;
+
+
+            var row = {
+                cartItemID: this.cartItems[i].CartItemID,
+                runID: this.cartItems[i].RunID,
+                location: this.cartItems[i].Location,
+                color: this.cartItems[i].Marker,
+                initialSize: this.cartItems[i].SizeMapID,
+                unitsPerPackage: this.cartItems[i].CountPerBatch,
+                quantity: this.cartItems[i].BatchCount,
+                dirty: false
+            }
+            productTable[index].rows.push(row);
```

Comment:
Slight confusion here. Why are we pushing something into a variable that will disappear as soon as this line finishes? The scope of productTable is this iteration of the for loop, correct?